django/cantusdb_project/main_app/management/commands/populate_exists_on_cantus_ultimus_field.py

```diff
@@ -0,0 +1,66 @@
+from main_app.models import Source
+from django.core.management.base import BaseCommand
+import requests
+from requests import Response
+
+# This management command updates the exists_on_cantus_ultimus field on a single
+# Source or all sources. If a source argument is provided, it will check if the
+# Source exists on Cantus Ultimus and update exists_on_cantus_ultimus to True if it
+# does.
+
+# If no Source argument is provided, it will update all Cantus Segment Sources in
+# the database.
+
+# run with `python manage.py add_cantus_ultimus_links <source_id>`
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "source_id",
+            nargs="?",
+            type=int,
+            default=None,
+            help="This command updates the exists_on_cantus_ultimus field for the Source with the specified ID",
+        )
+
+    def handle(self, *args, **kwargs):
+        source_id: int = kwargs["source_id"]
+        if source_id:  # update exists_on_cantus_ultimus for a specified source
+            try:
+                source: Source = Source.objects.get(id=source_id)
+                self.update_exists_on_cantus_ultimus(source)
+            except Source.DoesNotExist:
+                self.stdout.write(
+                    self.style.ERROR(f"Source {source_id} does not exist")
+                )
+                return
```

Comment:
Since we removed the else block we don't want the code under this if statement to run, and thus we need to exit the function early.

Reply:
I also think that if you catch an exception but don't raise another one, then it will continue.

Reply:
Oops, I was looking at a different part of the code.

Reply:
That return was pretty useful, though, since it made it explicit that it exited after writing the log message. Now it exits because it passes the `else`. Functionally they're the same, but in terms of readability it's useful to signal that the function ends there by returning. 

Also, if you ever add anything to the end of the handler, it will introduce a problem because it's not immediately obvious that it keeps executing after it passes that `except`.

Reply:
Good point, I'll leave it in then!